Content.Server/Automod/Actions/TextAutomodActionWarningPopup.cs

```diff
@@ -0,0 +1,35 @@
+﻿using System.Text;
+using Content.Server.Popups;
+using Content.Shared.Automod;
+using Content.Shared.Popups;
+using Robust.Shared.Player;
+
+namespace Content.Server.Automod.Actions;
+
+public sealed partial class TextAutomodActionWarningPopup : ITextAutomodAction
+{
+    [DataField]
+    public string Reason = "automod-action-warning-popup-reason";
+
+    public bool Skip(string fullText, Dictionary<string, int> patternMatches)
+    {
+        return false;
+    }
+
+    public bool RunAction(ICommonSession session,
+        string fullText,
+        Dictionary<string, int> patternMatches,
+        AutomodFilterDef filter,
+        string filterDisplayName,
+        IEntityManager entMan)
+    {
+        var str = Loc.GetString(
+            Reason,
+            ("matches", new StringBuilder().AppendJoin(", ", patternMatches.Keys)),
+            ("filterName", filterDisplayName));
+
+        entMan.System<PopupSystem>().PopupCursor(str, session, PopupType.LargeCaution);
```

Comment:
I am not sure why cursor popups make sense here. People are frequently not going to be looking at their cursor when typing a chat message.